src/pages/home/report/ReportActionCompose/SuggestionMention.tsx

```diff
@@ -270,9 +271,25 @@ function SuggestionMention(
                 }
 
                 return true;
-            });
+            }) as Array<PersonalDetails & {weight: number}>; // at this point we are sure that the details are not null
+
+            const sortedPersonalDetails = filteredPersonalDetails.sort((first, second) =>
```

Comment:
This feels overengineered. Let's put all the logic here, and if you want to use it in tests, export it from here. Let's also get rid of the eslint suppressions

Reply:
Yes to both